programs/kbuild/parse

```diff
@@ -1,19 +1,45 @@
 #!/usr/bin/env ruby
 
 # the workload of kbuild is defined as the iteration number
-runtime = 0
 iterations = 0
-start_time = 0
+real_seconds, user_seconds, sys_seconds = 0.0, 0.0, 0.0
+
 $stdin.each_line do |line|
-  start_time = $1.to_f if line =~ /^start time: (\d+\.\d+)/
-  if line =~ /^end time: (\d+\.\d+)/
-    runtime += $1.to_f - start_time
+  case line
+  when /real\s+(\d+h)?(\d+m)?(\d+\.\d+s)/
+    hours = $1.to_i
+    minutes = $2.to_i
+    seconds = $3.to_f
+    real_seconds += (hours * 3600) + (minutes * 60) + seconds
     iterations += 1
+  when /user\s+(\d+h)?(\d+m)?(\d+\.\d+s)/
+    hours = $1.to_i
+    minutes = $2.to_i
+    seconds = $3.to_f
+    user_seconds += (hours * 3600) + (minutes * 60) + seconds
+  when /sys\s+(\d+h)?(\d+m)?(\d+\.\d+s)/
+    hours = $1.to_i
+    minutes = $2.to_i
+    seconds = $3.to_f
+    sys_seconds += (hours * 3600) + (minutes * 60) + seconds
   end
 end
 
 if iterations != 0
-  buildtime_per_iteration = runtime / iterations
-  puts "buildtime_per_iteration: #{buildtime_per_iteration}"
+  average_real_hours = (real_seconds / iterations / 3600).to_i
+  average_real_minutes = ((real_seconds / iterations) % 3600 / 60).to_i
+  average_real_seconds = (real_seconds / iterations) % 60
+  puts "average_real: #{average_real_hours}h#{average_real_minutes}m#{average_real_seconds}s"
```

Comment:
similar to buildtime_per_iteration, here you can do
```
puts "real_time_per_iteration: #{real_seconds} / #{iterations}"
```

Reply:
ok

Reply:
After discussing with Yujie, the final time is shown in seconds to make it easier to make comparisons later for our project.